driver-core/src/main/com/mongodb/operation/OperationHelper.java

```diff
@@ -542,23 +542,19 @@ public void onResult(final T result, final Throwable t) {
         }
     }
 
-    static void withConnection(final AsyncWriteBinding binding, final AsyncCallableWithConnection callable) {
+    static void withConnectionAsync(final AsyncWriteBinding binding, final AsyncCallableWithConnection callable) {
```

Comment:
Generally we've been using "async" as a prefix rather than a suffix.  Let's follow the same pattern here and use `withAsyncConnection`.

Same for the rest

Reply:
Done